ggshield/sca/client.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List, Optional, Union, cast
+
+import marshmallow_dataclass
+import requests
+from pygitguardian.client import GGClient, _create_tar, is_ok, load_detail
+from pygitguardian.models import Base, BaseSchema, Detail, FromDictMixin
+
+from ggshield.sca.models import SCAScanParameters
+
+
+@dataclass
+class ComputeSCAFilesResult(Base, FromDictMixin):
+    sca_files: List[str]
+    potential_siblings: List[str]
+
+
+ComputeSCAFilesResult.SCHEMA = cast(
+    BaseSchema,
+    marshmallow_dataclass.class_schema(ComputeSCAFilesResult, base_schema=BaseSchema)(),
+)
+
+
+class DummyReturnModel(Base, FromDictMixin):
+    pass
+
+
+class SCAClient:
+    def __init__(self, client: GGClient):
+        self._client = client
+
+    @property
+    def base_uri(self):
+        return self._client.base_uri
+
+    def compute_sca_files(
+        self,
+        touched_files: List[str],
+        extra_headers: Optional[Dict[str, str]] = None,
+    ) -> Union[Detail, ComputeSCAFilesResult]:
+        response = self._client.post(
+            endpoint="sca/compute_sca_files/",
+            data={"touched_files": touched_files},
+            extra_headers=extra_headers,
+        )
+        result: Union[Detail, ComputeSCAFilesResult]
+        if is_ok(response):
+            result = ComputeSCAFilesResult.from_dict(response.json())
+        else:
+            result = load_detail(response)
+
+        result.status_code = response.status_code
+        return result
+
+    def sca_directory_scan_all(
+        self,
+        directory: Path,
+        filenames: List[str],
+        scan_parameters: SCAScanParameters,
+        extra_headers: Optional[Dict[str, str]] = None,
+    ):
+        """
+        Generates tar archive associated with filenames and launches
+        SCA scan via SCA public API.
+        """
+
+        tar = _create_tar(directory, filenames)
```

Comment:
Also, I'm not sure the tar creation part should belong to the client.
IMO the client should only be responsible for the communication with backend part.